cura/Machines/Models/CustomQualityProfilesDropDownMenuModel.py

```diff
@@ -1,31 +1,32 @@
-# Copyright (c) 2018 Ultimaker B.V.
+# Copyright (c) 2019 Ultimaker B.V.
 # Cura is released under the terms of the LGPLv3 or higher.
 
 from UM.Logger import Logger
 
+import cura.CuraApplication  # Imported this way to prevent circular references.
+from cura.Machines.ContainerTree import ContainerTree
 from cura.Machines.Models.QualityProfilesDropDownMenuModel import QualityProfilesDropDownMenuModel
 
-
-#
-# This model is used for the custom profile items in the profile drop down menu.
-#
+##  This model is used for the custom profile items in the profile drop down
+#   menu.
 class CustomQualityProfilesDropDownMenuModel(QualityProfilesDropDownMenuModel):
-
     def _update(self):
         Logger.log("d", "Updating {model_class_name}.".format(model_class_name = self.__class__.__name__))
 
-        active_global_stack = self._machine_manager.activeMachine
+        active_global_stack = cura.CuraApplication.CuraApplication.getInstance().getMachineManager().activeMachine
         if active_global_stack is None:
             self.setItems([])
             Logger.log("d", "No active GlobalStack, set %s as empty.", self.__class__.__name__)
             return
 
-        quality_changes_group_dict = self._quality_manager.getQualityChangesGroups(active_global_stack)
+        variant_names = [extruder.variant.getName() for extruder in active_global_stack.extruders.values()]
```

Comment:
I think you can use the `getCurrentQualityChangesGroups()` shortcut here.

Reply:
Yup. I think this was implemented before we made that shortcut. Fixed here: 7c90b5dd306629d3a3e99f38cfd9dc0a2760643f